projects/jsf-validation/src/lib/schema-validation.service.ts

```diff
@@ -111,6 +111,19 @@ export class SchemaValidationService {
 
     return null;
   }
+
+  private static isUriValid(uri: string): boolean {
+    if (!URI_VALID_CHARS_REGEX.test(uri)) {

```

Comment:
Why is this needed?

Reply:
The only downside to the `is.uri` package is that it allows non-ascii character in the uri. There are other validators that don't allow non-ascii characters, but they don't flag uris like `http://i-comms.truecommerce.net:42000AS2/F30510A3CH` as invalid. So I figured we will just have to check for invalid characters ourselves, since every other aspect of `is.uri` works as we want it.